### PR TITLE
Upgrade to rails 7.1

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

**Review by corylown — APPROVED** [review]
@jcoyne approved, but needs rebase

**Comment by jcoyne** [comment]
@corylown rebased.

File: Gemfile.lock [hunk 1]
```diff
@@ -443,10 +464,7 @@ GEM
     zeitwerk (2.6.12)
 
 PLATFORMS
-  x86_64-darwin-20
-  x86_64-darwin-21
   x86_64-darwin-22
-  x86_64-linux
```
Comment (corylown) [inline Gemfile.lock:449]: `x86_64-linux` was dropped in the rebase but we need it.

**Review by corylown — CHANGES_REQUESTED** [review]

File: Gemfile.lock (hunk 1 above)
Comment (jcoyne) [reply Gemfile.lock:449]: Got it, thanks.

**Review by corylown — APPROVED** [review]

**Pull request merged**